### PR TITLE
refactor(aws): make scaling policies customizable

### PR DESCRIPTION
**Pull request by anotherchrisberry**
Working on customizing the view of scaling policies internally at Netflix...

### CODE REVIEW

File: app/scripts/modules/amazon/src/domain/IScalingPolicy.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  'KilobitsSecond' | 'MegabitsSecond' | 'GigabitsSecond' | 'TerabitsSecond' | 'CountSecond' | 'None';
+
+export interface IStepAdjustment {
+  metricIntervalLowerBound?: number;
+  metricIntervalUpperBound?: number;
+  scalingAdjustment: number;
+}
+
+export interface IScalingPolicy {
+  policyARN?: string;
```
Comment (jrsquared) [inline app/scripts/modules/amazon/src/domain/IScalingPolicy.ts]: is this really optional?

File: app/scripts/modules/amazon/src/index.ts [hunk 2]
```diff
@@ -4,4 +4,8 @@ export * from './domain';
 
 export * from './keyPairs';
 
+export * from './templates';
```
Comment (jrsquared) [inline app/scripts/modules/amazon/src/index.ts]: alphabetical?

File: app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/upsert/alarm/metricSelector.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+describe('Component: metric selector', () => {
+
+  let cloudMetricsReader: CloudMetricsReader;
+  let $ctrl: MetricSelectorController;
+  let $componentController: IComponentControllerService;
+  let $q: IQService;
+  let $scope: IScope;
+  const alarmUpdated = new Subject<void>();
+
+  let alarm: IUpsertAlarmDescription, serverGroup: IServerGroup;
```
Comment (jrsquared) [inline app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/upsert/alarm/metricSelector.component.spec.ts]: put on separate lines like the `let`s above.

File: app/scripts/modules/amazon/src/domain/IScalingPolicy.ts (hunk 1 above)
Comment (anotherchrisberry) [inline app/scripts/modules/amazon/src/domain/IScalingPolicy.ts]: nope, good catch

**Pull request merged**